### PR TITLE
[PLAT-12231] Browser error correlation

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

Implement error correlation for browser performance

## Design

Call `addOnError` from bugsnag notifier when provided as a configuration option

## Changeset

Added new `bugsnag` configuration option to core client

## Testing

Added unit tests for adding span and trace id to a reported error

### CODE REVIEW

File: packages/core/lib/config.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
 import { type Plugin } from './plugin'
 import { isLogger, isNumber, isObject, isPluginArray, isString, isStringArray, isStringWithLength } from './validation'
 
+interface BugsnagErrorEvent {
+  setTraceCorrelation: (traceId: string, spanId: string) => void
```
Comment (imjoehaines) [inline packages/core/lib/config.ts:5]: I'd suggest marking this optional as it could be missing

```suggestion
  setTraceCorrelation?: (traceId: string, spanId: string) => void
```

File: packages/plugin-react-navigation/lib/create-navigation-container.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ export const createNavigationContainer: CreateNavigationContainer = (NavigationC
       }
 
       if (typeof onStateChange === 'function') {
-        onStateChange.apply(this, args)
+        onStateChange.apply(undefined, args)
```
Comment (imjoehaines) [inline packages/plugin-react-navigation/lib/create-navigation-container.tsx:30]: Is this change related?

**Review by imjoehaines — COMMENTED** [review]
Could we add a maze runner test for this?

File: packages/plugin-react-navigation/lib/create-navigation-container.tsx (hunk 2 above)
Comment (gingerbenw) [reply packages/plugin-react-navigation/lib/create-navigation-container.tsx:30]: no, actually! I can separate this out

**Comment by gingerbenw** [comment]
> Could we add a maze runner test for this?

Covered in PLAT-12233

File: packages/core/lib/core.ts [hunk 3]
```diff
@@ -68,6 +68,16 @@ export function createClient<S extends CoreSchema, C extends Configuration, T> (
     start: (config: C | string) => {
       const configuration = validateConfig<S, C>(config, options.schema)
 
+      if (configuration.bugsnag) {
```
Comment (imjoehaines) [inline packages/core/lib/core.ts:71]: Is it worth checking here that `setTraceCorrelation` exists so we avoid even adding the callback if the bugsnag-js version is too old?

**Review by imjoehaines — APPROVED** [review]

File: packages/core/lib/core.ts (hunk 3 above)
Comment (gingerbenw) [reply packages/core/lib/core.ts:71]: oooh, that's even better!

Comment (gingerbenw) [reply packages/core/lib/core.ts:71]: although, it appears I'm not entirely sure how to check for that...

Comment (imjoehaines) [reply packages/core/lib/core.ts:71]: I think you can check for `configuration.bugsnag.Event.prototype.setTraceCorrelation`:

https://github.com/bugsnag/bugsnag-js/blob/76f50d9b0684d8a445829c932fc182ee191775ab/packages/browser/src/notifier.js#L105

Comment (gingerbenw) [reply packages/core/lib/core.ts:71]: Do you think that would be best placed here, or in the config validation?

Comment (imjoehaines) [reply packages/core/lib/core.ts:71]: Sorry, missed this!

I think here is best — if/when we add more stuff that needs APIs from `Bugsnag`, we probably don't want to disable _all_ the error integrations if new APIs are missing

Comment (gingerbenw) [reply packages/core/lib/core.ts:71]: Raised PLAT-12350 to cover this, so we can get this ticket merged down
